lib/buffer.js

```diff
@@ -105,6 +105,10 @@ function fromObject(obj) {
     return b;
   }
 
+  if (obj == null) {
```

Comment:
@cjihrig Yes. If you look at the IR produced by Hydrogen it has a single instruction for checking "if-null-or-undefined".

@Fishrock123 `fromObject()` is a catch-all for other values not parsed elsewhere. This is an explicit check for the two following:
```
> new Buffer(null)
TypeError: Cannot read property 'buffer' of null

> new Buffer()
TypeError: Cannot read property 'buffer' of undefined
```
Everything else will fall through and hit the exception at the bottom of the function.